lib/libcheri/cheri_system.c

```diff
@@ -122,7 +122,8 @@ cheri_system_calloc(size_t count, size_t size,
 {
 	__capability void *ptr;
 
-	if ((ptr = (__capability void *)calloc(count, size)) == NULL)
+	/* XXXAR: bounds? */
```

Comment:
I think that when I wrote this code I assumed that the compiler would add bounds for casts from known allocators.  I'm not sure if that was ever true, but it was something we discussed.  I think we're going to want `calloc_c()` etc for hybrid mode with the `_c()` variants being aliases for the real deal in pure-capability mode.

Reply:
Sounds good, I can add those. It would also simplify some of the cases where I call cheri_csetbounds() on the result of calloc.